.clang-format

```diff
@@ -0,0 +1,11 @@
+BasedOnStyle: LLVM
+UseTab: Never
+IndentWidth: 4
+TabWidth: 4
+BreakBeforeBraces: Allman
+AllowShortIfStatementsOnASingleLine: false
+IndentCaseLabels: true
+ColumnLimit: 0
+AccessModifierOffset: -4
+PointerAlignment: Left
+ConstructorInitializerAllOnOneLineOrOnePerLine: true
```

Comment:
Yeah sorry about that. At first I really just wanted to add new options, but I kinda got carried away and ended up changing a whole bunch of stuff. I could try and separate this PR into multiple, but it'd take a while since I'd have to undo the changes and re-do them each in their own separate branches.

Reply:
nah, by now we've all read it 😁